### DISCUSSION TITLE
Existe alguma packages/método que faça essa função ?

### DISCUSSION THREAD

**Discussion by marcostd** [frontendbr/forum#1886] [Perguntas e Respostas]
Olá , estou iniciando meus estudos no Javascript e estou a procura de um packet, método ou estratégia que sane minha duvida, a qual eu não sei bem como formula-la diretamente. Mas vamos lá.

Estou aprendendo a programar e até me forço a escrever o código todo, evitando usar as ferramentas de alto completar. Deram me a dica de não usar tanto isso no inicio, pois poderia me atrapalhar. 

Mas existe uma série de estruturas repetitivas que , para mim, não faz sentido ficar rescrevendo-as sempre, já que são quase idênticas a anterior e não encontro meio automatizar essas "rotinas".

Veja o exemplo: 

```
pacienteTr.appendChild(montarTd(paciente.nome, "info-nome"));
pacienteTr.appendChild(montarTd(paciente.peso, "info-peso"));
pacienteTr.appendChild(montarTd(paciente.altura, "info-altura"));
pacienteTr.appendChild(montarTd(paciente.gordura, "info-gordura"));
pacienteTr.appendChild(montarTd(paciente.imc, "info-"));
```

Qualquer pessoa que leia apenas duas linhas desse código sabe qual o padrão que se segue e qual informação está faltando após o  **"info-"** . Queria saber se existe um shotcut, método ou package que melhore a performance nesse sentido?

**Comment by renedocket** [comment]
@marcoproject e ae beleza?

Pelo que eu entendi você gostaria de fazer processo apenas uma vez em relação a escrever esse código que que com a Td certo?

O que você poderia fazer na minha visão, seria criar uma função que recebe se esse dois valores **paciente e a info**.

exemplo:

_function adicionarInfoPaciente(caracteristica, descricaoInfo){
   pacienteTr.appendChild(montarTd(paciente.caracteristica, "info-"+descricaoInfo}));
}_

chamando a função você passaria exatamente os mesmo valores, só que reaproveitando o código.

_adicionarInfoPaciente(nome, nome);
adicionarInfoPaciente(peso, peso);
adicionarInfoPaciente(altura, altura);
adicionarInfoPaciente(gordura, gordura);
adicionarInfoPaciente(imc, imc);_

Acredito que seria isso pelo que você relatou. :)

**Comment by ojvribeiro** [comment]
E aí, @marcoproject!

Eu não sei qual lógica você tá tentando aplicar, mas tenho a mania de querer usar loops no que eu puder usar haha. 

Se você já sabe todas as informações que vai usar, você pode listar elas numa array de strings e rodar um `for` nela.

```javascript
let caracteristicas = ['nome', 'peso', 'altura', 'gordura', 'imc'];

function adicionarLinha() {
  for (let i in caracteristicas) {
    pacienteTr.appendChild(montarTd(paciente[caracteristicas[i]], `info-${caracteristicas[i]}`));
  }
}
```

Repare que no objeto `paciente` eu usei a _notação de colchetes_. Dessa forma eu consigo acessar o valor dele com as strings da array `caracteristicas`. Aí é só chamar `adicionarLinha()`.
